### PR TITLE
added category entity with repository, service and controller layers

### PR DESCRIPTION
**Pull request by dima-semeniuk**

### CODE REVIEW

**Review by Alexander1iubchenko — APPROVED** [review]
Looks cool. Good job

File: src/main/java/mate/academy/webapp/repository/book/BookRepository.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.data.jpa.repository.Query;
+
+public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+    @EntityGraph(attributePaths = "categories")
+    Page<Book> findAll(Pageable pageable);
+
+    @EntityGraph(attributePaths = "categories")
+    Optional<Book> findById(Long id);
+
+    @Query(value = "SELECT b FROM Book b JOIN FETCH b.categories c WHERE c.id = :categoryId")
```
Comment (vovasalyha) [inline src/main/java/mate/academy/webapp/repository/book/BookRepository.java:20]: Is it possible to use `@EntityGraph` here, and generate query by method name? I think it should work too.

File: src/main/java/mate/academy/webapp/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -20,49 +22,66 @@
...
+    public BookResponseDto updateById(Long id, CreateBookRequestDto requestDto) {
         bookRepository.findById(id).orElseThrow(
                 () -> new EntityNotFoundException("Can't find and update book by id: " + id)
         );
+        List<Category> categories = categoryRepository
+                .findAllById(requestDto.getCategories());
+        if (categories.isEmpty()) {
+            throw new EntityNotFoundException("No categories with id "
+                    + requestDto.getCategories());
+        }
```
Comment (vovasalyha) [inline src/main/java/mate/academy/webapp/service/impl/BookServiceImpl.java:69]: You have the same logic in 2 places. You can consider extracting it to private method (but it's optional, decide what is more suitable for you)

**Review by vovasalyha — APPROVED** [review]
GJ

**Pull request merged**